src/vma/dev/ring_bond.cpp

```diff
@@ -392,17 +395,24 @@ void ring_bond::inc_tx_retransmissions(ring_user_id_t id)
 
 bool ring_bond::reclaim_recv_buffers(descq_t *rx_reuse)
 {
-	devide_buffers_helper(rx_reuse, m_buffer_per_ring);
+	/* use this local array to avoid locking mechanizm
+	 * for threads synchronization. So every thread should use
+	 * own array. Set hardcoded number to meet C++11
+	 * VLA is not an official part of C++11.
+	 */
+	descq_t buffer_per_ring[MAX_NUM_RING_RESOURCES];
```

Comment:
In the  original code m_buffer_per_ring was created as:
```
m_buffer_per_ring = new descq_t[m_n_num_resources + 1];
```
while in your code you are creating buffer_per_ring array size smaller by 1:
```
descq_t buffer_per_ring[MAX_NUM_RING_RESOURCES];
```

Please explain why.